File: mobile/src/wirelessGecko/java/com/siliconlabs/bledemo/rangetest/RangeTestValues.java
package com.siliconlabs.bledemo.rangetest;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;
import java.util.List;

/**
 * @author Comarch S.A.
 */

public class RangeTestValues {

    public static final int PACKET_COUNT_REPEAT = 65535;

    public static final List<Integer> CHANNEL_LOOKUP;
    public static final List<Integer> PACKET_COUNT_LOOKUP;
    public static final List<Integer> ID_LOOKUP;

    private static final int CHANNELS_MIN = 0;
    private static final int CHANNELS_MAX = 19;

    private static final Integer[] PACKET_COUNTS = new Integer[]{500, 1000, 2500, 5000, 10000, 25000, 50000};

    private static final int ID_MIN = 0;
    private static final int ID_MAX = 32;

    static {
        int channelsLookupSize = CHANNELS_MAX - CHANNELS_MIN + 1;
        List<Integer> channelsLookup = new ArrayList<>(channelsLookupSize);
        for (int i = 0; i < channelsLookupSize; ++i) {
            channelsLookup.add(CHANNELS_MIN + i);
        }
        CHANNEL_LOOKUP = Collections.unmodifiableList(channelsLookup);

        PACKET_COUNT_LOOKUP = Collections.unmodifiableList(Arrays.asList(PACKET_COUNTS));

        int idLookupSize = ID_MAX - ID_MIN + 1;
        List<Integer> idLookup = new ArrayList<>(idLookupSize);
        for (int i = 0; i < idLookupSize; ++i) {
            idLookup.add(ID_MIN + i);
        }
        ID_LOOKUP = Collections.unmodifiableList(idLookup);
    }

    private RangeTestValues() {
    }
}

File: mobile/src/main/java/com/siliconlabs/bledemo/range_test/models/RangeTestValues.kt
package com.siliconlabs.bledemo.range_test.models

import java.util.*

/**
 * @author Comarch S.A.
 */
class RangeTestValues {
    companion object {
        const val PACKET_COUNT_REPEAT = 65535

        var CHANNEL_LOOKUP: List<Int>
        var PACKET_COUNT_LOOKUP: List<Int>
        var ID_LOOKUP: List<Int>

        private const val CHANNELS_MIN = 0
        private const val CHANNELS_MAX = 19

        private val PACKET_COUNTS = arrayOf(500, 1000, 2500, 5000, 10000, 25000, 50000)

        private const val ID_MIN = 0
        private const val ID_MAX = 32

        init {
            val channelsLookupSize = CHANNELS_MAX - CHANNELS_MIN + 1
            val channelsLookup: ArrayList<Int> = ArrayList(channelsLookupSize)
            for (i in 0 until channelsLookupSize) {
                channelsLookup.add(CHANNELS_MIN + i)
            }
            CHANNEL_LOOKUP = Collections.unmodifiableList(channelsLookup)

            PACKET_COUNT_LOOKUP = Collections.unmodifiableList(listOf(*PACKET_COUNTS))

            val idLookupSize = ID_MAX - ID_MIN + 1
            val idLookup: ArrayList<Int> = ArrayList(idLookupSize)
            for (i in 0 until idLookupSize) {
                idLookup.add(ID_MIN + i)
            }

            ID_LOOKUP = Collections.unmodifiableList(idLookup)
        }
    }
}